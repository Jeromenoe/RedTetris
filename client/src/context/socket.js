import React from 'react';
import socketio from "socket.io-client";


const localHost = "http://localhost:3001"; 
const herokuHost = "https://redtetris42-heroku.herokuapp.com"
const socketURL = process.env.NODE_ENV === "production" ? herokuHost : localHost

export const socket = socketio.connect(socketURL);
export const SocketContext = React.createContext();