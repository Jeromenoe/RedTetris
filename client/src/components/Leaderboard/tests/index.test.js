import React from 'react';
import { Leaderboard } from '../index';
import renderer from 'react-test-renderer';
import configureStore from "redux-mock-store";
import {Provider} from 'react-redux';

import Adapter from '@wojtekmaj/enzyme-adapter-react-17';
import { mount, configure } from 'enzyme';

import { SocketContext, socket } from "../../../context/socket"


configure({adapter: new Adapter()});



describe('Testing Leaderboard Component', () => {
	test('Check store isOver = true', async () => {
		const mockStore = configureStore();
		const initialState = {
			game: {
				isOver: true,
				players: [{
					rank: 1,
					name: "Jojo",
					score: 1000,
					rows: 10,
					level: 4
				},
				{
					rank: 2,
					name: "Joji",
					score: 1000,
					rows: 10,
					level: 4
				}]},
			me : {
				name : "Jojo"
			}
		};
		const store = mockStore(initialState);
		const tree = renderer.create(<Provider store={store}><Leaderboard /></Provider>).toJSON();
		expect(tree).toMatchSnapshot();
	  });

	test('Check store isOver = false', async () => {
		const mockStore = configureStore();
		const initialState = {
			game: {
				isOver: false,
				players: [{
					rank: 1,
					name: "Jojo",
					score: 1000,
					rows: 10,
					level: 4
				}]},
			me : {
				name : "Jojo"
			}
		};
		const store = mockStore(initialState);
		
		const tree = renderer.create(<Provider store={store}><Leaderboard /></Provider>).toJSON();
		expect(tree).toMatchSnapshot();
	});


	test('Test button leave game', async () => {
		const mockStore = configureStore();
		const initialState = {
			game: {
				isOver: false,
				players: [{
					rank: 1,
					name: "Jojo",
					score: 1000,
					rows: 10,
					level: 4
				}]},
			me : {
				name : "Jojo"
			}
		};
		const store = mockStore(initialState);
		
		const wrapper = mount(<Provider store={store}><SocketContext.Provider value={socket}><Leaderboard /></SocketContext.Provider></Provider>);
		wrapper.find('#btn-leave-in-game').simulate('click');
	});
	
});