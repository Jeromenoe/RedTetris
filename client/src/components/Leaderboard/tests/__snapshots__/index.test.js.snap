// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testing Leaderboard Component Check store isOver = false 1`] = `
<div
  className="sc-bdnxRM dZqNsG"
>
  <div
    className="sc-gtsrHT RaGih"
  >
    <strong
      style={
        Object {
          "alignSelf": "center",
          "borderBottom": "1px solid #ddd",
          "display": "block",
          "padding": "5px",
          "width": "90%",
        }
      }
    >
      Game Over
    </strong>
    <div
      className="player-score-container"
    >
      <div
        className="player-score-rank"
      >
        Rank
      </div>
      <div
        className="player-score-name"
      >
        Name
      </div>
      <div
        className="player-score-pts"
      >
        Score
      </div>
      <div
        className="player-score-lines"
      >
        Lines
      </div>
      <div
        className="player-score-lvl"
      >
        Level
      </div>
    </div>
    <div
      className="player-score-container"
      style={
        Object {
          "fontWeight": "bold",
        }
      }
    >
      <div
        className="player-score-rank"
      >
        1
      </div>
      <div
        className="player-score-name"
      >
        Jojo
         (me)
      </div>
      <div
        className="player-score-pts"
      >
        1000
      </div>
      <div
        className="player-score-lines"
      >
        10
      </div>
      <div
        className="player-score-lvl"
      >
        4
      </div>
    </div>
    <div
      className="btn-leaderboard-container"
    >
      <button
        className="btn-leave-leaderboard"
        onClick={[Function]}
      >
        Waiting Room
      </button>
      <button
        className="btn-leave-leaderboard"
        id="btn-leave-in-game"
        onClick={[Function]}
      >
        Leave
      </button>
    </div>
  </div>
</div>
`;

exports[`Testing Leaderboard Component Check store isOver = true 1`] = `
<div
  className="sc-bdnxRM kdmkhX"
>
  <div
    className="sc-gtsrHT RaGih"
  >
    <strong
      style={
        Object {
          "alignSelf": "center",
          "borderBottom": "1px solid #ddd",
          "display": "block",
          "padding": "5px",
          "width": "90%",
        }
      }
    >
      Game Over
    </strong>
    <div
      className="player-score-container"
    >
      <div
        className="player-score-rank"
      >
        Rank
      </div>
      <div
        className="player-score-name"
      >
        Name
      </div>
      <div
        className="player-score-pts"
      >
        Score
      </div>
      <div
        className="player-score-lines"
      >
        Lines
      </div>
      <div
        className="player-score-lvl"
      >
        Level
      </div>
    </div>
    <div
      className="player-score-container"
      style={
        Object {
          "fontWeight": "bold",
        }
      }
    >
      <div
        className="player-score-rank"
      >
        1
      </div>
      <div
        className="player-score-name"
      >
        Jojo
         (me)
      </div>
      <div
        className="player-score-pts"
      >
        1000
      </div>
      <div
        className="player-score-lines"
      >
        10
      </div>
      <div
        className="player-score-lvl"
      >
        4
      </div>
    </div>
    <div
      className="player-score-container"
      style={
        Object {
          "fontWeight": "normal",
        }
      }
    >
      <div
        className="player-score-rank"
      >
        2
      </div>
      <div
        className="player-score-name"
      >
        Joji
      </div>
      <div
        className="player-score-pts"
      >
        1000
      </div>
      <div
        className="player-score-lines"
      >
        10
      </div>
      <div
        className="player-score-lvl"
      >
        4
      </div>
    </div>
    <div
      className="btn-leaderboard-container"
    >
      <button
        className="btn-leave-leaderboard"
        onClick={[Function]}
      >
        Waiting Room
      </button>
      <button
        className="btn-leave-leaderboard"
        id="btn-leave-in-game"
        onClick={[Function]}
      >
        Leave
      </button>
    </div>
  </div>
</div>
`;
